File: frontend/src/types/booking.ts
export interface Passenger {
  id: number;
  name: string;
  email: string;
  seat_assignment: string | null;
}

export interface PassengerInput {
  name: string;
  email: string;
  seat_id?: number | null;
}

export interface BookingCreate {
  flight_id: number;
  passengers: PassengerInput[];
}

export interface Booking {
  id: number;
  user_id: number;
  flight_id: number;
  status: string;
  total_price: number;
  payment_status: string;
  created_at: string;
  passengers: Passenger[];
}

export interface BookingDetail extends Booking {
  flight_number: string | null;
  departure_airport: string | null;
  arrival_airport: string | null;
  departure_time: string | null;
  arrival_time: string | null;
}

File: backend/app/schemas/booking.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PassengerInput(BaseModel):
    name: str
    email: str
    seat_id: int | None = None


class BookingCreate(BaseModel):
    flight_id: int
    passengers: list[PassengerInput]


class PassengerResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    email: str
    seat_assignment: str | None = None


class BookingResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    flight_id: int
    status: str
    total_price: float
    payment_status: str
    created_at: datetime
    passengers: list[PassengerResponse] = []


class BookingDetail(BookingResponse):
    flight_number: str | None = None
    departure_airport: str | None = None
    arrival_airport: str | None = None
    departure_time: datetime | None = None
    arrival_time: datetime | None = None
